salt/states/x509.py

```diff
@@ -24,7 +24,7 @@
 
     peer:
       .*:
-        - x509.sign_remote_certificate
+        - sign_remote_certificate
```

Comment:
I tried the configuration this way but it refuses with `salt.exceptions.SaltInvocationError: ca_server did not respond salt master must permit peers to call the sign_remote_certificate function.`
When using `x509.sign_remote_certificate` everything works fine

Reply:
That's interesting. I had in the end both versions, with and without the `x509`, in the config. And I can confirm that now the x509 is required. I'm not sure if something changed or I made a mistake here. But I tested this pretty thoroughly back then.

So I guess we should revert this commit or update the documentation again.

Reply:
Created the PR: https://github.com/saltstack/salt/pull/57455